versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksFragment.java

```diff
@@ -0,0 +1,129 @@
+
+class RocksFragment extends RocksBaseValue<Fragment> implements Fragment, Evaluator {
+  static final String KEY_LIST = "keys";
+
+  private Stream<Key> keys;
+
+  RocksFragment() {
+    super();
+  }
+
+  @Override
+  public Fragment keys(Stream<Key> keys) {
+    this.keys = keys;
+    return this;
+  }
+
+  @Override
+  byte[] build() {
+    checkPresent(keys, KEY_LIST);
+    return ValueProtos.Fragment.newBuilder()
+      .setBase(buildBase())
+      .addAllKeys(keys.map(RocksBaseValue::buildKey).collect(Collectors.toList()))
+      .build()
+      .toByteArray();
+  }
+
+  /**
+   * Deserialize a RocksDB value into the given consumer.
+   *
+   * @param value the protobuf formatted value.
+   * @param consumer the consumer to put the value into.
+   */
+  static void toConsumer(byte[] value, Fragment consumer) {
+    try {
+      final ValueProtos.Fragment fragment = ValueProtos.Fragment.parseFrom(value);
+      setBase(consumer, fragment.getBase());
+      consumer.keys(fragment.getKeysList().stream().map(RocksBaseValue::createKey));
+    } catch (InvalidProtocolBufferException e) {
+      throw new StoreException("Corrupt Fragment value encountered when deserializing.", e);
+    }
+  }
+
+  @Override
+  public boolean evaluate(Condition condition) {
+    for (Function function: condition.getFunctionList()) {
+      // Retrieve entity at function.path
+      if (function.getPath().getRoot().isName()) {
+        ExpressionPath.NameSegment nameSegment = function.getPath().getRoot().asName();
+        final String segment = nameSegment.getName();
+        switch (segment) {
+          case ID:
+            if (!(!nameSegment.getChild().isPresent()
+                && function.getOperator().equals(Function.EQUALS)
+                && getId().toEntity().equals(function.getValue()))) {
+              return false;
+            }
+            break;
+          case KEY_LIST:
+            if (nameSegment.getChild().isPresent()) {
+              return false;
+            }
+            if (function.getOperator().equals(Function.EQUALS)) {
+              if (!keysAsEntityList(keys).equals(function.getValue())) {
+                return false;
+              }
+            } else if (function.getOperator().equals(Function.SIZE)) {
+              if (keys.count() != function.getValue().getNumber()) {
+                return false;
+              }
+            } else {
+              return false;
+            }
+            break;
+          default:
+            // Invalid Condition Function.
+            return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Produces an Entity List of Entity Lists for keys.
+   * Each key is represented as an entity list of string entities.
+   * @param keys stream of keys to convert
+   * @return an entity list of keys
+   */
+  private Entity keysAsEntityList(Stream<Key> keys) {
+    List<Key> keyListIn = keys.collect(Collectors.toList());
```

Comment:
Why not:

return Entity.ofList(keys.map(k -> Entity.ofList(k.getElements().stream().map(Entity::ofString))));

Reply:
Thanks. Updated.